### PR TITLE
Applied data fixes and adapted tests to check that sites are present …

### PR DESCRIPTION
**Pull request by supercasey**
…for data in the recent water year

### CODE REVIEW

**Comment by cschroed-usgs** [comment]
Thanks! This looks good!
I noticed that the expected sites are similar in many tests. Are some of them identical? Is there any way we could deduplicate the vectors of expected sites so that they are defined in fewer places? Or just one place?

One solution could be stashing the vectors in a global variable (or a few global variables) and adjusting the tests to use it/them.
Another solution could be moving all the most-recent-year tests into a single test file, and having all tests in the file rely on one (or a few) expected site vector(s) defined within that file.

Deduplicating this vector is most beneficial when we can make strong assumptions about the expected sites being the same in multiple use cases. I don't understand the data well enough to make those assumptions. Do you know if it is reasonable to assume that the expected sites will be the same across certain data types (loads vs flows)? Or across certain temporal densities (monthly vs annual)?

File: tests/testthat/test_annual_load.R [hunk 1]
```diff
@@ -108,9 +108,9 @@ test_that("There are no duplicate values", {
 })
 
 test_that("Most recent water year has all of the necessary sites ", {
-  temp_aloads_recent<-temp_aloads[temp_aloads$WY %in% max(temp_aloads$WY),] 
+  temp_aloads_recent<-temp_aloads[temp_aloads$WY%in%max(temp_aloads$WY),] 
   expected <- sort(c("HAZL","PADU","GRAN","CLIN","WAPE","KEOS","VALL","GRAF","SIDN","OMAH","ELKH","LOUI","DESO","HERM","THEB","SEDG","HARR","LITT","KERS",
                      "STFR","BATO","BELL","MELV","CALU","MORG","SUMN","STTH","GULF","NEWH","CANN","MISS"))
-  actual <- sort(unique(temp_aloads_recent$SITE_ABB))
+  actual <- sort(unique(temp_aloads_recent[temp_aloads_recent$SITE_ABB%in%expected,"SITE_ABB"]))
```
Comment (cschroed-usgs) [inline tests/testthat/test_annual_load.R]: Interesting! Why the change from
```
actual <- sort(unique(temp_aloads_recent$SITE_ABB))
```
to...
```
actual <- sort(unique(temp_aloads_recent[temp_aloads_recent$SITE_ABB%in%expected,"SITE_ABB"]))
```

?

Comment (cschroed-usgs) [reply tests/testthat/test_annual_load.R]: I think they produce the same results. Does the more-complicated version produce better output on test failure?

Comment (cschroed-usgs) [reply tests/testthat/test_annual_load.R]: The corresponding `mflow` test is using the simpler version. If the more complicated version is better, should the `mflow` test be using the more complicated version as well?

Comment (supercasey) [reply tests/testthat/test_annual_load.R]: I changed the form because the sites in the network, or sites that have the data needed to compute loads, can change from year to year, this applies more for the annual load, annual flow, and daily flow files that have nearly all the sites than for the monthly load and flow files, which have only a subset of sites.  The change in the test only checks to make sure the set of sites listed is present in the data, not the other way around.  The sites listed are important sites in the Mississippi River Basin currently, but I'll probably add some more sites outside the basin to the annual and daily files in the future.  I will make the change to the mflow test for consistency.

Comment (cschroed-usgs) [reply tests/testthat/test_annual_load.R]: That makes sense. Thanks!

**Comment by cschroed-usgs** [comment]
I approved this without merging in case you still wanted to perform any of the deduplication that I brought up in a previous comment. If it's not possible, or relevant to deduplicate, you are welcome to merge!
